src/js/form/form-validate.js

```diff
@@ -0,0 +1,72 @@
+import { validateText } from './utils/validate-text';
+import { validateLength } from './utils/validate-length';
+import { validateEmail } from './utils/validate-email';
+import { validatePhone } from './utils/validate-phone';
+import { requireError } from './constant';
+
+const formElement = document.querySelector('.js-form');
+const formInputElements = document.querySelectorAll('.js-form-input');
+const button = document.querySelector('.js-form-submit');
+
+function validate(target) {
+    if (target.required && !target.value.length) {
+        return [requireError]
+    }
+
+    switch(target.dataset.name) {
+        case 'name':
+            return [validateText(target.value), validateLength(25, target.value)].filter(Boolean)
+
+        case 'company':
+            return [validateText(target.value), validateLength(50, target.value)].filter(Boolean)
+
+        case 'phone':
+            return [validatePhone(target.value)].filter(Boolean)
+
+        case 'email':
+            return [validateEmail(target.value)].filter(Boolean)
+
+        case 'message':
+            return [validateLength(250, target.value)].filter(Boolean)
+        
+    }
+}
+
+function toggleButton(toggle) {
+    button.disabled = toggle;
+    button.classList.toggle('js-button', toggle);
+}
+
+function checkButton() {
+    const isFormError = Boolean(document.querySelector('.form__error_show'));
+
+    toggleButton(isFormError);
+}
+
+function validateForm(target) {
+    const errors = validate(target);
+    const isError = errors.length;
+    const parentElement = target.parentElement;
+    const errorElement = parentElement.querySelector('.form__error');
+
+    target.classList.toggle('form__item_error', isError);
+
+    errorElement.textContent = isError ? errors[errors.length - 1] : '';
+    errorElement.classList.toggle('form__error_show', isError);
+
+    checkButton();
+}
+
+formInputElements.forEach(el => {
+    el.addEventListener('blur', ({target}) => {
+        validateForm(target);
+
+        target.dataset.touch = "true";
```

Comment:
use consistent quotes
```suggestion
        target.dataset.touch = 'true';
```